components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/workflow/WorkflowExtensionsConfigBuilder.java

```diff
@@ -20,166 +20,32 @@
         workflowConfig = null;
     }
-    public static void build(WorkflowConfig defaultConfig) {
-        Optional<String> workflowConfigFileContent = readFile(WORKFLOW_CONFIG_YML);
-
-        if (workflowConfigFileContent.isPresent()) {
-
-            Constructor constructor = new Constructor(WorkflowConfig.class);
-            TypeDescription workflowConfigType = new TypeDescription(WorkflowConfig.class);
-            constructor.addTypeDescription(workflowConfigType);
-
-            TypeDescription wfexecutorInfo = new TypeDescription(WorkflowExecutorInfo.class);
-            wfexecutorInfo.putListPropertyType("property", WorkflowConfigProperties.class);
-            constructor.addTypeDescription(wfexecutorInfo);
-
-            Yaml yaml = new Yaml(constructor);
-            workflowConfig = (WorkflowConfig) yaml.load(workflowConfigFileContent.get());
-            if (logger.isDebugEnabled()) {
-                logger.debug("WorkflowConfiguration file content: " + workflowConfigFileContent.get());
-                logger.debug("WorkflowConfig  object :" + workflowConfig.toString());
-            }
-
-        } else {
-            logger.warn("Could not find the workflow-extensions.yml in conf location. Using default configurations");
-            workflowConfig = defaultConfig;
-        }
-    }
-
-    /**
-     * Read file content to a String. The optional will have the file content from the given file in a {@link String}
-     * when the file exists in the system or when it is found in the classpath.
-     *
-     * @param fileName The file name
-     * @return An optional {@link String}
-     */
-    private static Optional<String> readFile(final String fileName) {
-        Optional<File> configFile = getConfigFile(fileName);
-        try (final InputStream in = configFile.isPresent() ? new FileInputStream(configFile.get())
-                : Thread.currentThread().getContextClassLoader().getResourceAsStream(fileName)) {
-            if (in != null) {
-                try (BufferedReader buffer = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8));
-                        Stream<String> stream = buffer.lines()) {
-                    String fileContent = stream.map(Utils::substituteVariables)
-                            .collect(Collectors.joining(System.lineSeparator()));
-                    return Optional.of(fileContent);
-                }
-            }
-        } catch (IOException e) {
-            throw new RuntimeException("Failed to read lines from the file: " + fileName, e);
+    public static void build(ConfigProvider configProvider) {
+
+        try {
+            workflowConfig = configProvider.getConfigurationObject(WorkflowConfig.class);
+        } catch (CarbonConfigurationException e) {
+            logger.error("Error while loading the configuration for worlflow ", e);
+            //build default executors
+            workflowConfig = new WorkflowConfig();
```

Comment:
this is a default configuration for workflow. if deployment.yaml config loading issue happens, then we will use the defautl workflow config